src/main/java/htsjdk/samtools/SAMRecordDuplicateComparator.java

```diff
@@ -57,27 +60,44 @@
     
     public SAMRecordDuplicateComparator() {}
 
+    @Deprecated // not thread safe. will be removed in future version
```

Comment:
The problem isn't that it's not thread-safe (I think), it's that the eventual order depends on the order of the headers in the list.....right?